metrics/lib/features/dashboard/presentation/state/project_metrics_store.dart

```diff
@@ -1,74 +1,140 @@
+import 'package:metrics/features/dashboard/domain/usecases/receive_poject_updates.dart';
 import 'package:metrics/features/dashboard/presentation/model/build_result_bar_data.dart';
+import 'package:metrics/features/dashboard/presentation/model/project_metrics.dart';
+import 'package:rxdart/rxdart.dart';
 
 /// The store for the project metrics.
 ///
-/// Stores the [Coverage] and [BuildMetrics].
+/// Stores the [Project]s and its [BuildMetrics].
 class ProjectMetricsStore {
-  final GetProjectCoverage _getCoverage;
-  final GetBuildMetrics _getBuildMetrics;
-  List<Point<int>> _projectPerformanceMetrics;
-  List<Point<int>> _projectBuildNumberMetrics;
-  List<BuildResultBarData> _projectBuildResultMetrics;
-  BuildMetrics _buildMetrics;
-  Coverage _coverage;
+  static const int defaultNumberOfBuildResults = 14;
+  static const Duration defaultBuildMetricsLoadingPeriod = Duration(days: 7);
+
+  final ReceiveProjectUpdates _receiveProjectsUpdates;
+  final ReceiveBuildMetricsUpdates _receiveBuildMetricsUpdates;
+  final Map<String, StreamSubscription> _buildMetricsSubscriptions = {};
+  final BehaviorSubject<Map<String, ProjectMetrics>> _projectsMetricsSubject =
+      BehaviorSubject();
+
+  StreamSubscription _projectsSubscription;
 
   /// Creates the project metrics store.
   ///
   /// The [_getCoverage] and [_getBuildMetrics] use cases should not be null.
-  ProjectMetricsStore(this._getCoverage, this._getBuildMetrics)
-      : assert(
-          _getCoverage != null && _getBuildMetrics != null,
+  ProjectMetricsStore(
+    this._receiveProjectsUpdates,
+    this._receiveBuildMetricsUpdates,
+  ) : assert(
+          _receiveProjectsUpdates != null &&
+              _receiveBuildMetricsUpdates != null,
           'The use cases should not be null',
         );
 
-  Coverage get coverage => _coverage;
+  Stream<List<ProjectMetrics>> get projectsMetrics =>
+      _projectsMetricsSubject.map((metricsMap) => metricsMap.values.toList());
+
+  /// Subscribes to projects and its metrics.
+  Future<void> subscribeToProjects() async {
+    final projectsStream = _receiveProjectsUpdates();
+    await _projectsSubscription?.cancel();
 
-  List<Point<int>> get projectPerformanceMetrics => _projectPerformanceMetrics;
+    _projectsSubscription = projectsStream.listen((projects) {
+      if (projects == null || projects.isEmpty) {
+        _projectsMetricsSubject.add({});
+        return;
+      }
 
-  List<Point<int>> get projectBuildNumberMetrics => _projectBuildNumberMetrics;
+      final projectsMetrics = _projectsMetricsSubject.value ?? {};
 
-  List<BuildResultBarData> get projectBuildResultMetrics =>
-      _projectBuildResultMetrics;
+      final newProjectsIds = projects.map((project) => project.id);
+      projectsMetrics.removeWhere((projectId, value) {
+        final remove = !newProjectsIds.contains(projectId);
+        if (remove) _buildMetricsSubscriptions[projectId]?.cancel();
```

Comment:
Shouldn't we remove a subscription from _buildMetricsSubscriptions after it's cancelled?